src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/ConvertData/ConvertDataEngineTests.cs

```diff
@@ -68,37 +91,65 @@ public async Task GivenConvertDataRequest_WithDefaultTemplates_CorrectResultShou
         public async Task GivenConvertDataRequest_WithUnconfiguredRegistry_ContainerRegistryNotConfiguredExceptionShouldBeThrown(string templateReference)
         {
             var convertDataEngine = GetDefaultEngine();
-            var request = GetHl7V2RequestWithTemplateReference(templateReference);
-            await Assert.ThrowsAsync<ContainerRegistryNotConfiguredException>(() => convertDataEngine.Process(request, CancellationToken.None));
+            var hl7v2Request = GetHl7V2RequestWithTemplateReference(templateReference);
+            await Assert.ThrowsAsync<ContainerRegistryNotConfiguredException>(() => convertDataEngine.Process(hl7v2Request, CancellationToken.None));
+
+            var ccdaRequest = GetCcdaRequestWithTemplateReference(templateReference);
+            await Assert.ThrowsAsync<ContainerRegistryNotConfiguredException>(() => convertDataEngine.Process(ccdaRequest, CancellationToken.None));
         }
 
         [Theory]
         [InlineData("       ")]
         [InlineData("Abc")]
         [InlineData("¶Š™œãý£¾")]
         [InlineData("MSH|")]
-        public async Task GivenConvertDataRequest_WithWrongInputData_ConvertDataFailedExceptionShouldBeThrown(string inputData)
+        public async Task GivenHl7V2ConvertDataRequest_WithWrongInputData_ConvertDataFailedExceptionShouldBeThrown(string inputData)
         {
             var convertDataEngine = GetDefaultEngine();
             var request = GetHl7V2RequestWithInputData(inputData);
             var exception = await Assert.ThrowsAsync<ConvertDataFailedException>(() => convertDataEngine.Process(request, CancellationToken.None));
             Assert.True(exception.InnerException is DataParseException);
         }
 
+        [Theory]
+        [InlineData("       ")]
+        [InlineData("Abc")]
+        [InlineData("¶Š™œãý£¾")]
+        [InlineData("<?xml version=\"1.0\"?><?xml-stylesheet type='text/xsl' href=''?>")]
+        public async Task GivenCcdaConvertDataRequest_WithWrongInputData_ConvertDataFailedExceptionShouldBeThrown(string inputData)
+        {
+            var convertDataEngine = GetDefaultEngine();
+            var request = GetCcdaRequestWithInputData(inputData);
+            var exception = await Assert.ThrowsAsync<ConvertDataFailedException>(() => convertDataEngine.Process(request, CancellationToken.None));
```

Comment:
>ConvertDataFailedException [](start = 53, length = 26)

I don't see that exception in `OperationOutcomeExceptionFilterAttribute`.
What we return to customer in case of bad data? InternalServerError?

Reply:
We will return 400 Bad Request as this exception captures user errors like invalid input. The ```ConvertDataUnhandledException ``` will capture unexpecture expection and return ```500 Internal Server Error```

Reply:
My bad, for some reason I thought default behavior for `OperationOutcomeExceptionFilterAttribute` to return 500. I would still suggest to expilcitly specify that exception in filter, but you right it returns 400 and there is test for that in E2E tests.

---
In reply to: [600156327](https://github.com/microsoft/fhir-server/pull/1756#discussion_r600156327) [](ancestors = 600156327)

Reply:
Agree. We can add exception handling to ```OperationOutcomeExceptionFilterAttribute``` explicitly to avoid issue in future updates.

Reply:
Added `ConvertDataUnhandledException `  to `OperationOutcomeExceptionFilterAttribute `.